### PR TITLE
some code cleanup to remove some hacked code

### PR DESCRIPTION
**Pull request by ayoub-benali**
Introduced a builder for the Application Context

### CODE REVIEW

File: src/main/scala/de/m7w3/signal/ApplicationContext.scala [hunk 1]
```diff
@@ -38,11 +46,33 @@ case class ApplicationContext(config: Config.SignalDesktopConfig) {
...
+  def setStore(s: SignalDesktopProtocolStore): ApplicationContextBuilder = {
+    store = s
+    this
+  }
+
+  def setAccount(a: AccountHelper): ApplicationContextBuilder = {
+    account = a
+    this
   }
+  def build(): InitiatedContext = InitiatedContext(account, store)
```
Comment (mfelsche) [inline src/main/scala/de/m7w3/signal/ApplicationContext.scala]: require initialized args

File: src/main/scala/de/m7w3/signal/ApplicationContext.scala [hunk 2]
```diff
@@ -38,11 +46,33 @@ case class ApplicationContext(config: Config.SignalDesktopConfig) {
         SignalDesktopProtocolStore(dBActionRunner)
     }
   }
+}
 
-  def tryLoadExistingStore(password: String, skipCache: Boolean): Try[SignalDesktopProtocolStore] = {
-    databaseContext.dbActionRunner(password, onlyIfExists = true, skipCache)
-      .map(SignalDesktopProtocolStore(_))
+case class InitiatedContext(account: AccountHelper, store: SignalDesktopProtocolStore) extends ApplicationContext {
+
```
Comment (mfelsche) [inline src/main/scala/de/m7w3/signal/ApplicationContext.scala]: i need a `DatabaseContext` here

**Review by mfelsche — COMMENTED** [review]
good start, carry on

File: src/main/scala/de/m7w3/signal/ApplicationContext.scala (hunk 2 above)
Comment (ayoub-benali) [reply src/main/scala/de/m7w3/signal/ApplicationContext.scala]: fixed

File: src/main/scala/de/m7w3/signal/ApplicationContext.scala (hunk 1 above)
Comment (ayoub-benali) [reply src/main/scala/de/m7w3/signal/ApplicationContext.scala]: fixed

**Comment by ayoub-benali** [comment]
@mfelsche would say ready to merge

**Review by mfelsche — APPROVED** [review]
do it

**Pull request merged**